instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientAttributesExtractor.java

```diff
@@ -54,7 +54,15 @@ public static <REQUEST, RESPONSE> HttpClientAttributesExtractorBuilder<REQUEST,
   @Override
   public void onStart(AttributesBuilder attributes, Context parentContext, REQUEST request) {
     super.onStart(attributes, parentContext, request);
-    internalSet(attributes, SemanticAttributes.HTTP_URL, getter.url(request));
+    internalSet(attributes, SemanticAttributes.HTTP_URL, stripSensitiveData(getter.url(request)));
+  }
+
+  private static String stripSensitiveData(String url) {
+    if (url.isEmpty() || url == null) {
```

Comment:
It should be the other way:
```suggestion
    if (url == null || url.isEmpty()) {
```
If `url` were null, the `url.isEmpty()` check would fail with an NPE.

Reply:
You are right, now idea how this happened :sweat: